### DISCUSSION TITLE
Problems while running MCMC for a 'semi-detached' model.

### DISCUSSION THREAD

**Discussion by tilakpawar** [phoebe-project/phoebe2#709] [Questions]
Has anyone tried setting up MCMC runs or just a 'run_compute' from a distribution for the case of a semi-detached system (requiv@secondary = requiv_max@secondary; determined by the mass ratio) ? I am getting a 'secondary overflowing at periastron' error and the run terminates. I had expected that it will accept the physically acceptable values and reject the rest. Is there a particular way I need to set-up the distributions for such systems so as to avoid the error?

**Comment by kecnry** [comment]
Are you using the [semidetached constraint](http://phoebe-project.org/docs/latest/tutorials/requiv_crit_semidetached) to ensure they're exactly equivalent or just setting `requiv` manually?  

If you're not using the constraint, are you sampling over any parameters that affect `requiv_max` (and therefore once any of those are sampled and `requiv_max` shrinks at all from your original value, that will result in an overflow).  Can you run a single instance of `run_compute` with your set values or does that fail as well?  There should be some allowed tolerance where internally it will just set them to the max value... if that isn't working, that might be a bug (please send a minimal script or bundle so we can try to reproduce).

**Reply by tilakpawar** [reply]
I use the constraint to make the secondary fill its roche lobe entirely. requiv_primary was set to an initial value and was then optimised later, with other parameters. I am able to initiate a bundle and also optimise this system without any issues. The problem occurs while trying the compute a model using distributions.

I attach a minimum script to reproduce this error and the associated bundle.
[run_compute_issues.zip](https://github.com/phoebe-project/phoebe2/files/10852052/run_compute_issues.zip)

**Reply by kecnry** [reply]
Ok, it sounds to me like you're doing everything correct from your end, so we'll try to reproduce and see if this is a bug that can be fixed.  Thanks for reporting it!

EDIT: I created #711 for the bug so it doesn't get lost here in a discussion.

**Reply by tilakpawar** [reply]
Thanks! :)
I'll keep an eye on #711 and work on the other 'well-behaved' cases till then.
